client/src/shared/utils/interface.ts

```diff
@@ -159,3 +164,45 @@ export interface Learner {
   firstname: string;
   lastname: string;
 }
+
+export interface CourseLesson {
+  id: number;
+  title: string;
+  link: string;
+  description: string;
+  is_active: boolean;
+  created_at: string;
+  updated_at: string;
+  course: number;
+}
+
+export interface CourseLearner {
+  trainee_id: number;
+  user_id: number;
+  fullname: string;
+  firstname: string;
+  lastname: string;
+  email: string;
+  progress: number;
+}
+
+export interface CourseCollection {
+  id: number;
+  categories: CourseCategory[];
+  lessons: CourseLesson[];
+  learners: CourseLearner[];
+  code: string;
+  name: string;
+  description: string | null;
+  img_path: string | null;
+  preview_vid_path: string | null;
+  created_at: string | null;
+  updated_at: string | null;
+  company: number;
+  author: number;
+  status: number;
```

Comment:
since we do not have final ERD yet, please send only required fields from BE and create an interface accordingly, for example, I think we will be using `company`, `status`,  `preview_vid_path`. or `code` on FE